File: posts/ai-server/test_websocket.py
#!/usr/bin/env python3
"""
WebSocket Test for Posts-AI Service
Tests the /ws/chats/:id endpoint
"""

import asyncio
import websockets
import json
import uuid
from datetime import datetime


async def test_websocket():
    chat_id = str(uuid.uuid4())
    user_id = str(uuid.uuid4())

    uri = f"ws://localhost:3014/ws/chats/{chat_id}?user_id={user_id}"

    print("\n=== WebSocket Test for Posts-AI Service ===")
    print(f"Service URL: ws://localhost:3014")
    print(f"Chat ID: {chat_id}")
    print(f"User ID: {user_id}")
    print(f"\nConnecting to WebSocket...\n")

    try:
        async with websockets.connect(uri) as websocket:
            print("✓ WebSocket connected successfully!")
            print(f"  Server: {websocket.remote_address}")
            print(
                f"  Connection time: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")

            # Send test message
            test_message = {
                "prompt": "How to write effective blog posts?",
                "agent": "economist"
            }

            print(f"\nSending test message:")
            print(f"  {json.dumps(test_message, indent=2)}")

            await websocket.send(json.dumps(test_message))

            # Wait for response with timeout
            print(f"\nWaiting for response...")
            try:
                response = await asyncio.wait_for(websocket.recv(), timeout=5.0)
                print(f"\n✓ Received message from backend:")
                try:
                    parsed = json.loads(response)
                    print(f"  {json.dumps(parsed, indent=2)}")
                except json.JSONDecodeError:
                    print(f"  {response}")

            except asyncio.TimeoutError:
                print(f"  (No response received within 5 seconds)")
                print(f"  This is expected if AI service is not configured")

            print(f"\n✓ WebSocket test completed")

    except ConnectionRefusedError:
        print(f"✗ Connection refused - service not running on localhost:3014")
    except Exception as e:
        print(f"✗ Error: {type(e).__name__}: {e}")


if __name__ == "__main__":
    asyncio.run(test_websocket())

File: posts/ai-server/test-websocket.js
#!/usr/bin/env node

const WebSocket = require('ws')
const { v4: uuidv4 } = require('uuid')

const chatId = uuidv4()
const userId = uuidv4()

console.log('\n=== WebSocket Test for Posts-AI Service ===')
console.log(`Service URL: ws://localhost:3014`)
console.log(`Chat ID: ${chatId}`)
console.log(`User ID: ${userId}`)
console.log(`\nConnecting to WebSocket...`)

const ws = new WebSocket(
  `ws://localhost:3014/ws/chats/${chatId}?user_id=${userId}`,
)

ws.on('open', function open() {
  console.log('✓ WebSocket connected successfully!')
  console.log('\nSending test message...')

  const testMessage = {
    prompt: 'How to write effective blog posts?',
    agent: 'economist',
  }

  console.log(`Message: ${JSON.stringify(testMessage, null, 2)}`)
  ws.send(JSON.stringify(testMessage))
})

ws.on('message', function message(data) {
  console.log(`\nReceived message from backend:`)
  try {
    const parsed = JSON.parse(data)
    console.log(JSON.stringify(parsed, null, 2))
  } catch (e) {
    console.log(data)
  }
})

ws.on('error', function error(err) {
  console.error('✗ WebSocket Error:', err.message)
})

ws.on('close', function close() {
  console.log('\n✓ WebSocket closed')
  process.exit(0)
})

// Auto-close after 10 seconds
setTimeout(() => {
  console.log('\n--- Test timeout (10s) ---')
  ws.close()
}, 10000)
